cli/flox/src/commands/environment.rs

```diff
@@ -783,8 +783,47 @@ pub struct Push {
 }
 
 impl Push {
-    pub async fn handle(self, flox: Flox) -> Result<()> {
+    pub async fn handle(self, mut flox: Flox) -> Result<()> {
         subcommand_metric!("push");
+
+        if flox.floxhub_token.is_none() {
+            let options = indoc! {"
+                * login to floxhub with 'flox auth login',
+                * set the 'floxhub_token' field to '<your token>' in your config
+                * set the '$FLOX_FLOXHUB_TOKEN=<your_token>' environment variable."};
```

Comment:
this makes error messages a bit wordy, but at least tells you what to do:

```
ERROR: You are not logged in to floxhub.
    
    Can not automatically login to floxhub in non-interactive context.
    
    To login you can either
    * login to floxhub with 'flox auth login',
    * set the 'floxhub_token' field to '<your token>' in your config
    * set the '$FLOX_FLOXHUB_TOKEN=<your_token>' environment variable.
```

We can also just refer to the config or auth manpage which might be a bit more concise